nengo/spa/vocab.py

```diff
@@ -81,6 +81,10 @@ def __init__(self, dimensions, randomize=True, unitary=False,
         self.include_pairs = include_pairs
         self._identity = None
         self.rng = rng
+        self.read_only = False
+
+        # The parent of a non-subset vocabulary is itself
+        self.parent = self
```

Comment:
It's optional and [not guaranteed](https://docs.python.org/2/reference/datamodel.html#objects-values-and-types). It's not so much the garbage collection that I'm worried about, it just seems like an antipattern to me so my eyebrow instinctively raises. If there's absolutely no other way to do this, then so be it, but I suspect there's another way that ends up being simpler and easier.